server/data/data-library/src/test/java/org/apache/james/rrt/lib/RecipientRewriteTableContract.java

```diff
@@ -411,6 +411,19 @@ default void listSourcesShouldReturnWhenMultipleSourceMapping() throws Exception
         assertThat(virtualUserTable().listSources(mapping)).contains(source, source2);
     }
 
+    @Test
+    default void listSourcesShouldReturnWhenSourceHasMultipleMappings() throws Exception {
```

Comment:
Let's do also something for `getStoredMappings` ?

Reply:
I'm not sure what you mean. This was a bug I stumbled upon in the previous code while working on this, so I added a test just to make sure it no longer occurs in the new design. Everything else I assume should be covered by existing tests...

Reply:
With the export import procedure via webadmin it may not be relevant.

Let's keep as is.